database/src/main/java/com/pip/chatbot/repository/fortune/StockRepository.java

```diff
@@ -1,60 +1,56 @@
 package com.pip.chatbot.repository.fortune;
 
 import com.pip.chatbot.model.finance.Stock;
-import com.pip.chatbot.model.finance.Symbol;
+import lombok.AllArgsConstructor;
+import org.jooq.DSLContext;
 
-import java.time.LocalDate;
-import java.util.ArrayList;
-import java.util.Arrays;
+import java.time.LocalDateTime;
 import java.util.List;
 import java.util.Optional;
 
+import static com.pip.chatbot.jooq.fortune.tables.Stock.STOCK;
+
+@AllArgsConstructor
 public class StockRepository {
-    List<Stock> stocks;
-
-    public StockRepository(){
-            stocks = Arrays.asList(
-                    new Stock(1, new Symbol("EURUSD=X", "Euro to USD", true), (float) 1.1078, LocalDate.parse("2020-05-29"), true),
-                    new Stock(2, new Symbol("EURUSD=X", "Euro to USD", true), (float) 1.1115, LocalDate.parse("2020-06-01"), true),
-                    new Stock(3, new Symbol("EURUSD=X", "Euro to USD", true), (float) 1.1135, LocalDate.parse("2020-06-02"), true),
-                    new Stock(4, new Symbol("OTGLF", "CD Projekt S.A.", false), (float) 100.75, LocalDate.parse("2020-05-29"), true),
-                    new Stock(5, new Symbol("OTGLF", "CD Projekt S.A.", false), (float) 96.50, LocalDate.parse("2020-06-01"), true),
-                    new Stock(6, new Symbol("OTGLF", "CD Projekt S.A.", false), (float) 98.08, LocalDate.parse("2020-06-02"), true));
+
+    private DSLContext dsl;
+
+    public Optional<Stock> get(Long id) {
+        return dsl.selectFrom(STOCK)
+                .where(STOCK.ID.eq(id))
+                .fetchOptionalInto(Stock.class);
     }
 
-    public Optional<Stock> get(int id) {
-        Stock resultStock = null;
-        for (Stock stock : stocks) {
-            if (stock.getId() == id)
-                resultStock = stock;
-        }
-        return Optional.ofNullable(resultStock);
+    public Stock createStock(Stock stock) {
+        return dsl.insertInto(STOCK)
+                .set(STOCK.ID, stock.getId())
+                .set(STOCK.SYMBOL, stock.getSymbol())
+                .set(STOCK.VALUE, stock.getValue())
+                .set(STOCK.DATE, stock.getDate())
+                .set(STOCK.IS_HISTORICAL, stock.isHistorical())
+                .returning()
+                .fetchOne()
+                .into(Stock.class);
     }
 
-    public Optional<Stock> getCurrencyForDay(String symbol, LocalDate date) {
-        Stock resultStock = null;
-        for (Stock stock : stocks) {
-            if (stock.getSymbol().getSymbol().equals(symbol) && stock.getDate().getDayOfMonth() == date.getDayOfMonth())
-                resultStock = stock;
-        }
-        return Optional.ofNullable(resultStock);
+    public Optional<Stock> getCurrencyForDay(String symbol, LocalDateTime date) {
+        return dsl.selectFrom(STOCK)
+                .where(STOCK.SYMBOL.eq(symbol))
+                .and(STOCK.DATE.eq(date))
```

Comment:
W schemie mamy date jako timestamp, a tego nie potrzebujemy i utrudnia nam to życie.
1.Zróbmy migrację, która zmienia type date z timestamp na date.
2.Wygenerujmy jeszcze raz kod jooq.
3.Używajmy LocalDate, jak poprzednio